pkg/cloudprovider/provider/azure/provider.go

```diff
@@ -827,6 +834,10 @@ func (p *provider) Validate(spec v1alpha1.MachineSpec) error {
 		return errors.New("resourceGroup is missing")
 	}
 
+	if c.VNetResourceGroup == "" {
```

Comment:
This is not needed as you are already setting the `VNetResourceGroup` in here https://github.com/medmedchiheb/machine-controller/blob/azure-vnet-resource-group/pkg/cloudprovider/provider/azure/provider.go#L224